xdsl/dialects/llvm.py

```diff
@@ -372,7 +408,7 @@ def verify(self):
 class ArithmeticBinOperation(IRDLOperation, ABC):
     """Class for arithmetic binary operations."""
 
-    T: ClassVar = VarConstraint("T", BaseAttr(IntegerType))
+    T: ClassVar = VarConstraint("T", LLVMCompatibleSignlessIntegerConstraint)
```

Comment:
I would say that the changes of this kind, for ints and floats, are quite easy  to review, and would be great to merge separately, so we can have a higher-quality discussion. Also, does this work?
```suggestion
    T: ClassVar = VarConstraint("T", IntegerType | VectorType[IntegerType])
```